src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```diff
@@ -76,31 +70,55 @@ trait Contexts { self: Analyzer =>
 
   var lastAccessCheckDetails: String = ""
 
-  /** List of symbols to import from in a root context.  Typically that
-   *  is `java.lang`, `scala`, and [[scala.Predef]], in that order.  Exceptions:
+  lazy val generalRootImports: List[Import] =
+    settings.Ypredef.value
+      .map(global.newUnitParser(_, "<flag-Ypredef>").parseRule(_.importExpr()))
+      .collect { case imp: Import =>
+        // I can't seem to figure out how to use the freshly parsed tree...
+        // Instead we get to recreate it
+        gen.mkImportFromSelector(
```

Comment:
See the inline code comments. The parsed tree seems to be lacking symbols on the import's expr tree. I couldn't figure out how to assign them without throwing errors.